icu4c/source/common/ucnvmbcs.cpp

```diff
@@ -540,12 +540,12 @@ const UConverterSharedData _MBCSData={
 static const uint32_t
 gb18030Ranges[14][4]={
     {0x10000, 0x10FFFF, LINEAR(0x90308130), LINEAR(0xE3329A35)},
-    {0x9FA6, 0xD7FF, LINEAR(0x82358F33), LINEAR(0x8336C738)},
+    {0x9FBC, 0xD7FF, LINEAR(0x82359135), LINEAR(0x8336C738)}, // for -2022 replace beginning of range with explicit mappings
```

Comment:
How about changiing
```
-  {0x9FA6, 0xD7FF, LINEAR(0x82358F33), LINEAR(0x8336C738)},
+ {0x9FA6, 0x9FB3, LINEAR(0x82358F33), LINEAR(0x83369036)},
+ {0x9FBC, 0xD7FF, LINEAR(0x82359135), LINEAR(0x8336C738)},
```
and not adding the following
```
<U9FA6> \x82\x35\x8F\x33 |0
<U9FA7> \x82\x35\x8F\x34 |0
<U9FA8> \x82\x35\x8F\x35 |0
<U9FA9> \x82\x35\x8F\x36 |0
<U9FAA> \x82\x35\x8F\x37 |0
<U9FAB> \x82\x35\x8F\x38 |0
<U9FAC> \x82\x35\x8F\x39 |0
<U9FAD> \x82\x35\x90\x30 |0
<U9FAE> \x82\x35\x90\x31 |0
<U9FAF> \x82\x35\x90\x32 |0
<U9FB0> \x82\x35\x90\x33 |0
<U9FB1> \x82\x35\x90\x34 |0
<U9FB2> \x82\x35\x90\x35 |0
<U9FB3> \x82\x35\x90\x36 |0
```
to source/data/mappings/gb18030.ucm ?

Reply:
Since the ranges have to be checked before any access to the explicit mappings, it seemed like adding another range would have more perf impact than just adding a few explicit mappings (if the range is just for 10 or 20 chars or so)

Reply:
sure, but there are binary size impact.

Reply:
Please do not change the algorithmic ranges.
- It is not necessary to change them, because they are used only in the error handling functions when there are no explicit mappings in the .ucm file.
- It would be bad to change them, because the code would then no longer work with the original conversion tables.

Reply:
My mistake, I thought the algorithmic ranges were checked first before the explicit mappings.